### ISSUE TITLE
Task fails with incomplete folder check

### ISSUE THREAD

**Issue by phptek** [silverstripe/silverstripe-assets#352]
On line 103 of `SecureAssetsMigrationHelper.php`, the query for the `Folder` `DataObject` subclass naively assumes a record will always be returned, which of course is rarely always the case. Therefore a guard should be added before calling the method (any method) `getFilename()` on a potentially `null` query result:

FROM:

`$migratedPath = $this->migrateFolder($filesystem, $folder->getFilename());`

TO:

```
/** @var Folder $folder */
if (!$folder = Folder::get()->byID($id)) {
    $this->logger->info('Folder not found, skipping');
    continue;
}

$migratedPath = $this->migrateFolder($filesystem, $folder->getFilename());
```

**Comment by maxime-rainville** [comment]
Thanks for point this out. I see you got some related commits ... do you fancy doing a PR?

**Comment by robbieaverill** [comment]
Fixed in 1.5.x - thanks @phptek!
